ignite/templates/app/files/cmd/{{binaryNamePrefix}}d/cmd/root.go.plush

```diff
@@ -121,13 +123,38 @@ func NewRootCmd() *cobra.Command {
 		flags.FlagKeyringBackend: "test",
 	})
 
+	autoCliOpts, err := enrichAutoCliOpts(autoCliOpts, clientCtx)
+	if err != nil {
+		panic(err)
+	}
+
 	if err := autoCliOpts.EnhanceRootCommand(rootCmd); err != nil {
 		panic(err)
 	}
 
 	return rootCmd
 }
 
+func enrichAutoCliOpts(autoCliOpts autocli.AppOptions, clientCtx client.Context) (autocli.AppOptions, error) {
+	autoCliOpts.AddressCodec = addresscodec.NewBech32Codec(sdk.GetConfig().GetBech32AccountAddrPrefix())
+	autoCliOpts.ValidatorAddressCodec = addresscodec.NewBech32Codec(sdk.GetConfig().GetBech32ValidatorAddrPrefix())
+	autoCliOpts.ConsensusAddressCodec = addresscodec.NewBech32Codec(sdk.GetConfig().GetBech32ConsensusAddrPrefix())
+
+	var err error
+	clientCtx, err = config.ReadFromClientConfig(clientCtx)
```

Comment:
This reintroduce the bug that if you change the home flag, the keyring won't work

Reply:
Worked for this PR; you can test the example in the description. And the integration tests for `tx` are working now

Reply:
https://github.com/ignite/cli/actions/runs/8304630600/job/22730513199?pr=4026

Reply:
Okay, then I think we should need to be change ReadDefaultConfigFromClientConfig by this in the provider and rollback all the rest.